pkg/util/general/healthz.go

```diff
@@ -19,71 +19,84 @@ package general
 import (
 	"fmt"
 	"sync"
-
-	"k8s.io/klog/v2"
+	"time"
 )
 
-var healthzCheckRules sync.Map
+var healthzCheckMap sync.Map
 
 // HealthzCheckName describes which rule name for this check
 type HealthzCheckName string
 
 // HealthzCheckState describes the checking results
 type HealthzCheckState string
 
-type HealthzCheckResponse struct {
-	State   HealthzCheckState `json:"state"`
-	Message string            `json:"message"`
+type HealthzCheckStatus struct {
+	State          HealthzCheckState `json:"state"`
+	Message        string            `json:"message"`
+	LastUpdateTime time.Time         `json:"lastUpdateTime"`
+	// when LastUpdateTime is not updated for more than TimeoutPeriod, we consider this rule is failed.
+	// 0 or negative value means no need to check the LastUpdateTime.
+	TimeoutPeriod      time.Duration `json:"timeoutPeriod"`
+	UnhealthyStartTime time.Time     `json:"unhealthyStartTime"`
+	// when current State is not HealthzCheckStateReady, and it lasts more than GracePeriod, we consider this rule is failed.
+	// 0 or negative value means no need to check the UnhealthyStartTime.
+	GracePeriod time.Duration `json:"gracePeriod"`
```

Comment:
actually, `GracePeriod` is usually used in a `positive` way, i.e. to check the duration for healthy state rather than healthy state, and makes it kind of intricate if we don't infer to the comments carefully; maybe `TolerationPeriod` will be better

Reply:
not insist though

Reply:
agree with you.